client/src/components/createCredentialDialog/createCredentialDialog.js

```diff
@@ -56,38 +57,96 @@ class CreateCredentialDialog extends React.Component {
 
     this.sshKeyFileValidator = new RegExp(/^\/.*$/);
 
-    helpers.bindMethods(this, ['cancel', 'save', 'addResultsCallback']);
+    helpers.bindMethods(this, [
+      'cancel',
+      'save',
+      'addResultsCallback',
+      'updateResultsCallback'
+    ]);
   }
 
   componentWillReceiveProps(nextProps) {
     if (!this.props.show && nextProps.show) {
-      this.resetIntialState();
-      this.props.getCredentials();
+      this.resetIntialState(nextProps);
     }
   }
 
-  resetIntialState() {
-    this.setState({
-      credentialName: '',
-      authorizationType: 'usernamePassword',
-      sshKeyFile: '',
-      passphrase: '',
-      username: '',
-      password: '',
-      becomeMethod: 'sudo',
-      becomeUser: '',
-      becomePassword: '',
-      credentialNameError: '',
-      usernameError: '',
-      sskKeyFileError: '',
-      becomeUserError: ''
-    });
+  resetIntialState(nextProps) {
+    if (nextProps && nextProps.editing && nextProps.editCredential) {
+      this.setState({
+        credentialName: nextProps.editCredential.name,
+        authorizationType:
+          nextProps.editCredential.ssh_keyfile &&
+          nextProps.editCredential.ssh_keyfile.length
+            ? 'usernamePassword'
+            : 'sshKey',
+        sshKeyFile: nextProps.editCredential.ssh_keyfile,
+        passphrase: nextProps.editCredential.passphrase,
+        username: nextProps.editCredential.username,
+        password: nextProps.editCredential.password,
+        becomeMethod: nextProps.editCredential.become_method,
+        becomeUser: nextProps.editCredential.become_user,
+        becomePassword: nextProps.editCredential.become_password,
+        credentialNameError: '',
+        usernameError: '',
+        sskKeyFileError: '',
+        becomeUserError: ''
+      });
+    } else {
+      this.setState({
+        credentialName: '',
+        authorizationType: 'usernamePassword',
+        sshKeyFile: '',
+        passphrase: '',
+        username: '',
+        password: '',
+        becomeMethod: 'sudo',
+        becomeUser: '',
+        becomePassword: '',
+        credentialNameError: '',
+        usernameError: '',
+        sskKeyFileError: '',
+        becomeUserError: ''
+      });
+      this.props.getCredentials();
+    }
```

Comment:
Note: I'll take a look at investigating a default reset action for these ```Store.dispatch({type: credentialsTypes.CREDENTIAL_RESET})``` might help cut down on some of this